File: app/src/main/java/com/example/testesantander/login/LoginContract.java
package com.example.testesantander.login;

import android.content.Context;

import com.example.testesantander.domain.User;

public class LoginContract {
    public interface View{

        void navigateToList(User user );

        Context getActivity();

        void enableButton(boolean b);

        void errorPassword(String s);

        void errorUsername(String s);

        void showProgress(final boolean show);


    }
    public interface Presenter{

        void login(String username, String password);

        boolean validUsername(String user);

        boolean validPassword(String password);

    }
}

File: BankApp/app/src/main/java/br/com/vinicius/bankapp/ui/login/LoginContract.kt
package br.com.vinicius.bankapp.ui.login

import androidx.appcompat.app.AppCompatActivity
import br.com.vinicius.bankapp.domain.user.User
import com.google.android.material.snackbar.Snackbar

class LoginContract {

    interface View {
        fun notification(message: String)

        fun saveUserPreferences(user: User)

        fun showProgressBar(show: Boolean)

        fun getActivity(): AppCompatActivity

        fun showSnack(message: String)
    }

    interface Presenter {
        val view: LoginContract.View

        fun startLogin(username: String, password:String)

        fun showSnack(view: android.view.View, message: String, length: Int = Snackbar.LENGTH_LONG)
    }
}